app/assets/javascripts/test/tracing-view-screenshot-tests/dataset_rendering_helpers.js

```diff
@@ -0,0 +1,64 @@
+/* eslint import/no-extraneous-dependencies: ["error", {"peerDependencies": true}], no-await-in-loop: 0 */
+const fetch = require("node-fetch");
+const pixelmatch = require("pixelmatch");
+
+const DEV_AUTH_TOKEN =
+  "815aeacaec3392040b5232557599362e93f2155f942cb17ca75de51b954f3835f364bc5df0ab323807136942acae8cbaca8451c4e0240e011bcce68cfd05afb41f1baf37b9f36d4cc87cdd44cc221046ac720233f247e0bc5adb90a512cb4f405093d6c4552bbe86dc12cdb77458fa760385cb9690d81df5435334b58ab901dd";
+
+async function createExplorational(datasetName, typ, withFallback, branchName) {
+  const url = `https://${branchName}.webknossos.xyz/api/datasets/${datasetName}/createExplorational`;
+  return (await fetch(url, {
+    body: JSON.stringify({ typ, withFallback }),
+    method: "POST",
+    headers: {
+      "X-Auth-Token": DEV_AUTH_TOKEN,
+      "Content-Type": "application/json",
+    },
+  })).json();
+}
+
+async function screenshotDataset(page, branchName, datasetName) {
+  const createdExplorational = await createExplorational(
+    datasetName,
+    "skeleton",
+    false,
+    branchName,
+  );
+  return openTracingViewAndScreenshot(page, branchName, createdExplorational.id);
+}
+
+async function openTracingViewAndScreenshot(page, branchName, annotationId) {
+  await page.goto(
+    `https://${branchName}.webknossos.xyz/annotations/Explorational/${annotationId}`,
+    { timeout: 0 },
+  );
+
+  let canvas;
+  while (canvas == null) {
```

Comment:
Is there a general timeout for the tests? Or do you have to kill the process if this hangs for some reason?

Reply:
ava can be configured to shut down the tests if there was no test completed in the last x seconds/minutes. I thought this was configured globally, but it is actually not, I'll add a timeout of 5 minutes for the screenshot tests.